### PR TITLE
Add vulnerabilities use case page

### LINKED ISSUE #5172: Fixing Vulnerabilities Landing Page
Note: Nutanix's quote won't link to a case study, not live yet

The [Figma file](https://www.figma.com/file/aA1cNEm2JWtBSuijZyxROs/Use-Case-Template?node-id=0%3A1) is the source of truth for copy and design

🔗Figma file → WIP tab, look for V2
🔗[Meta](https://docs.google.com/spreadsheets/d/1_mZxd_RLysQ0VMUzHK_gQZ8Jak-o7cTyaWS8tqYLZvg/edit#gid=149103332)
🔗[Creative files](https://drive.google.com/drive/folders/1_TpGTK2bt9NJ6PSBSaMab67q_zgkVHj0)

CTAs

- Request a demo: https://info.sourcegraph.com/demo-request
- Try Sourcegraph: https://about.sourcegraph.com/get-started
- Log4j was the tip of the iceberg / Read the blog post: https://about.sourcegraph.com/blog/log4j-log4shell-0-day/
- Source: Security intelligence: https://securityintelligence.com/news/news-vulnerabilities-25-days-remediate/
- Code Search: https://about.sourcegraph.com/code-search/
- Batch Changes: https://about.sourcegraph.com/batch-changes/
- Code Monitoring: https://docs.sourcegraph.com/code_monitoring
- Clodflare case study: https://about.sourcegraph.com/case-studies/cloudflare-accelerates-debugging-and-improves-security
- Indeed case study: https://about.sourcegraph.com/case-studies/indeed-accelerates-development-velocity
- Explore other use cases: https://about.sourcegraph.com/use-cases/
- Log4j Log4Shell 0-day: find, fix, and track affected code: https://about.sourcegraph.com/blog/log4j-log4shell-0-day/
- The Nine Circles of Dependency Hell (and a roadmap out): https://about.sourcegraph.com/blog/nine-circles-of-dependency-hell/
- How to remove secrets from your codebase: https://about.sourcegraph.com/blog/eliminate-secrets-from-codebase-with-universal-code-search/
- Ready to get started? https://about.sourcegraph.com/get-started

### PR DESCRIPTION
**Pull request by katjuell**
Closes #5172 by adding a `vulnerabilities` page under a new `use-cases` pages directory. Also moves `pages/use-cases.tsx` to `pages/use-cases/index.tsx`.

## What are the acceptance criteria? 
- Page matches design specs 
- ^^ on mobile
- All meta is correct
- Carousel behavior everywhere is correct (`code-insights`, `cloud-beta`, this page. We use this component on all of these pages, so any changes we make to it need to look okay in all of these places).
- `BackButton` on get started pages looks okay (that was updated and turned into a shared component).
- Adds link to vulnerabilities page from main index page.
Screen Shot 2022-03-15 at 5 28 04 PM

## To Test
- Check out the branch and/ or deploy preview and navigate to `/use-cases/vulnerabilities`.
- Check the above criteria.

## Notes
- We are using Material Design icons here, since this is the icon library we use across our properties. The designs use [Phosphor](https://phosphoricons.com/), but CPT has determined that we will not be using icons from this library until we are ready to make a site-wide change. Until that point, we will be using a single icon library for the sake of consistency.
- The `BlogListItem` section behavior on mobile (`Related resources` section) matches the behavior of that component on `code-insights` on mobile, which CPT would like to keep consistent between the pages (using `flex-column` and full-width images on mobile).
- The gradient background on the quote carousel section now matches what's in #5179 — again, for consistency.

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1201682886788502

### CODE REVIEW

File: website/src/components/BackButton.tsx [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import { Link } from 'gatsby'
+import ArrowLeftIcon from 'mdi-react/ArrowLeftIcon'
+import React, { FunctionComponent } from 'react'
+
+export const BackButton: FunctionComponent<{ href: string; text: string; bold: boolean }> = ({ href, text, bold }) => (
```
Comment (katjuell) [inline website/src/components/BackButton.tsx:5]: Moved this out so we could share it across components and pages!

File: website/src/components/QuoteCarousel.tsx [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+}
+
+export const QuoteCarousel: FunctionComponent<QuoteCarouselProps> = ({ items, autoAdvance, className }) => {
+    const carouselHook = useCarousel(items, autoAdvance ?? false)
+    const carouselItems = carouselHook.carouselItems.items as Blockquote[]
+    const carouselStyles = 'd-flex flex-lg-row flex-column justify-content-center text-center'
+
+    return (
+        <div className={className ? classNames(carouselStyles, className) : classNames(carouselStyles)}>
+            <div className="d-lg-flex d-none align-items-center">
```
Comment (katjuell) [inline website/src/components/QuoteCarousel.tsx:34]: The [designs for this page](https://www.figma.com/file/aA1cNEm2JWtBSuijZyxROs/Fixing-vulnerabilities-page?node-id=177%3A2) call for this behavior (`d-lg-flex`/ `d-none`), which means that there's not a way to manually advance between panels on mobile (autoadvance will work). This seems ... odd to me?

File: website/src/components/product/CustomerLogosSectionAnimated.tsx [hunk 3]
```diff
@@ -177,9 +177,10 @@ interface Props {
     showButton: boolean
     className: String
     showSection: boolean
+    noCta?: boolean
```
Comment (katjuell) [inline website/src/components/product/CustomerLogosSectionAnimated.tsx:180]: This seemed like the least disruptive way of getting what I needed (no extra section and no CTA buttons anywhere on desktop or mobile), but open to suggestions.

File: website/src/components/CustomCarousel.tsx [hunk 4]
```diff
@@ -19,16 +19,16 @@ interface CarouselProps {
...
     buttonLabel: string
     headerClass: string
     text: string | ReactNode
     itemClass: string
 }
 
 const CustomCarousel: FunctionComponent<CarouselProps> = props => {
-    const carouselMainStyles = 'd-flex flex-wrap'
-    const carouselRightPanelStyles = 'col-lg-6 col-md-8 col-sm-12 mt-lg-5 ml-md-6 px-lg-0'
+    const carouselMainStyles = 'd-flex flex-wrap flex-lg-row flex-column'
```
Comment (bretthayes) [inline website/src/components/CustomCarousel.tsx:30]: I noticed that the carousel arrows push the page layout horizontally from the `sm` to `md` breakpoints. You might need to modify this a bit.

Screen Shot 2022-03-16 at 2 50 41 PM

**Review by bretthayes — CHANGES_REQUESTED** [review]
Nice work! Left some comments and feedback below.

File: website/src/pages/use-cases/vulnerabilities.tsx [hunk 5]
```diff
@@ -0,0 +1,396 @@
...
+]
+
+const UseCasePage: FunctionComponent<PageProps> = props => (
+    <Layout
+        location={props.location}
+        meta={{
+            title: 'Find and fix security vulnerabilities',
+            description:
+                'Search across all your repositories to find and resolve vulnerabilities in minutes, not days.',
+            image: 'https://about.sourcegraph.com/sourcegraph-og.png',
```
[the remark this answers is not in the document]
Comment (katjuell) [reply website/src/pages/use-cases/vulnerabilities.tsx:188]: I was going off of the meta listed in the [spreadsheet](https://docs.google.com/spreadsheets/d/1_mZxd_RLysQ0VMUzHK_gQZ8Jak-o7cTyaWS8tqYLZvg/edit#gid=149103332)

File: website/src/components/CustomCarousel.tsx (hunk 4 above)
Comment (katjuell) [reply website/src/components/CustomCarousel.tsx:30]: Weird. Why am I not seeing that?
Screen Shot 2022-03-17 at 3 58 28 PM

File: website/src/components/QuoteCarousel.tsx (hunk 2 above)
Comment (katjuell) [reply website/src/components/QuoteCarousel.tsx:34]: Sweet, yeah, I'm gonna add that behavior then. I will also add a note to the top-level PR description that we've decided to add that functionality.

File: website/src/pages/use-cases/vulnerabilities.tsx [hunk 6]
```diff
@@ -0,0 +1,396 @@
...
+const CarouselItem: FunctionComponent<{ header: string; text: ReactNode }> = ({ header, text }) => (
+    <>
+        <h2>{header}</h2>
+        {text}
+    </>
+)
+
+const items = [
+    {
+        id: 0,
```
[the remark this answers is not in the document]
Comment (bretthayes) [reply website/src/pages/use-cases/vulnerabilities.tsx:25]: Amazing!! 😄

File: website/src/components/CustomCarousel.tsx (hunk 4 above)
Comment (bretthayes) [reply website/src/components/CustomCarousel.tsx:30]: Weird, I'm not sure 🤷🏻‍♂️  I thought I was cached but I hard refreshed and also tried in incognito and I can still reproduce.  Check out [this loom](https://www.loom.com/share/fd751290063d4fddaeb4d20520a590a1)

Comment (katjuell) [reply website/src/components/CustomCarousel.tsx:30]: I ended up updating those styles for other reasons, so marking this as resolved for now.

File: website/src/components/CustomCarousel.tsx [hunk 7]
```diff
@@ -15,20 +15,23 @@ interface CarouselProps {
     previousItem?: CarouselItem
     currentItemIndex?: number
     autoAdvance?: boolean
+    hideCarouselNav?: boolean
```
Comment (katjuell) [inline website/src/components/CustomCarousel.tsx:18]: Honestly, we may need to break this carousel into two when we port. There are enough deviations of behavior between designs of the "same" component for this to already feel incredibly brittle.

**Comment by katjuell** [comment]
@bretthayes there may still be some copy edits — currently trying to confirm that changes are closed. Going off of the "V3" design in the [Figma](https://www.figma.com/file/aA1cNEm2JWtBSuijZyxROs/Fixing-vulnerabilities-page?node-id=0%3A1)

**Comment by marybelzer** [comment]
@katjuell got word that the figma was the final version and approved - I shared the updated link with stakeholders for final review!

File: website/src/components/CustomCarousel.tsx (hunk 7 above)
Comment (bretthayes) [reply website/src/components/CustomCarousel.tsx:18]: Yeah I like that idea! Given how this component has evolved, I'd suggest we have a Vertical and a Horizontal carousel component and collocate some related logic into sibling files.

I like the rule of thumb of keeping tightly knit/scoped logic close by in sibling files and then if we have shared logic that get used in other components, then we break that logic out into the higher level directories like the higher level shared hooks folder.

So something like this seems more convenient:
```
components/
  Carousel/
    Horizontal.tsx
    Vertical.tsx
    index.tsx
    move.ts (the carousel hook)
    ...whatever else we need or could clean up into another sibling file
```

Comment (katjuell) [reply website/src/components/CustomCarousel.tsx:18]: Yeah, that sounds good to me! I can port over all the pages with carousels in one [issue](https://github.com/sourcegraph/about-replatform/issues/39) in the replatform and that will be a good chance for a rewrite

File: website/src/css/styles.scss [hunk 8]
```diff
@@ -357,6 +356,60 @@ table {
...
+// Custom heights
+$heights: 100, 200, 300, 350, 400, 450, 500, 550, 600, 650, 700, 750, 800, 850;
+
+// Custom height utility classes
+@each $height in $heights {
+    .height-#{$height} {
+        height: #{$height}px;
+    }
+    @media (min-width: 1200px) {
+        .height-x-lg-#{$height} {
```
Comment (bretthayes) [inline website/src/css/styles.scss:368]: Is `x-lg` meaning extra large? If so, should be `xl` instead

File: website/src/css/styles.scss [hunk 9]
```diff
@@ -357,6 +356,60 @@ table {
...
+            height: #{$height}px;
+        }
+    }
+}
+
+.height-auto {
+    height: auto;
+}
+@media (min-width: 1200px) {
+    .height-x-lg-auto {
```
Comment (bretthayes) [inline website/src/css/styles.scss:393]: `xl`

File: website/src/pages/use-cases/index.tsx [hunk 10]
```diff
@@ -17,9 +19,9 @@ export default ((props: any) => (
                 'See how the most productive dev teams use Sourcegraph to build software you rely on. From remediating vulnerabilities to streamlining code reuse, our customers use Sourcegraph to solve big code problems.',
             image: 'https://about.sourcegraph.com/sourcegraph-og.png',
         }}
-        heroAndHeaderClassName="use-cases-page__header-and-hero"
+        heroAndHeaderClassName={`${styles.headerAndHero}`}
```
Comment (bretthayes) [inline website/src/pages/use-cases/index.tsx:22]: You can omit the template string

**Review by bretthayes — CHANGES_REQUESTED** [review]
Just a few small changes and then you're good to go!

**Pull request merged**